vector/src/main/java/fr/gouv/tchap/activity/NotificationListenerDetectionActivity.java

```diff
@@ -0,0 +1,133 @@
+
+        String[] large_descriptions = flat.split(":");
+        String[] appname_array = new String[large_descriptions.length];
+
+        for (int i = 0; i < large_descriptions.length; i++) {
+            String[] splitted = large_descriptions[i].split("/");
+            String packagename = splitted[0];
+
+            appname_array[i]=getAppName(packagename, context);
+        }
+
+        return appname_array;
+    }
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        StringBuilder string_builder = null;
+        String message  = "";
+
+        String[] notificationListenerNames = new String[]{};
+
+        if (!PreferencesManager.detectNotificationListener(this)) {
+            finish();
+            return;
+        }
+
+
+        notificationListenerNames = getListeningServiceAppName(getApplicationContext(), getContentResolver());
+
+        if (notificationListenerNames.length == 0) {
+            finish();
+            return;
+        }
+
+        string_builder = new StringBuilder();
+
+        message  = getResources().getString(R.string.notification_security_dialog_message);
+
+
+        for (String s : notificationListenerNames) {
+            string_builder.append("\n").append("- ").append(s);
+        }
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+
+        builder.setMessage(message + string_builder.toString())
+                .setIcon(R.drawable.message_notification_transparent)
+                .setTitle(R.string.security_dialog_title)
+                .setPositiveButton(R.string.security_dialog_start, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        finish();
+                    }
+                }).setNeutralButton(R.string.security_dialog_start_an_stop_detecting, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                PreferencesManager.putDetectNotificationListener(NotificationListenerDetectionActivity.this, false);
+                finish();
+            }
+        }).setNegativeButton(R.string.security_dialog_stop, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                m_finish_app = true;
+                finish();
+            }
+        })
+                .setOnCancelListener(new DialogInterface.OnCancelListener() {
+                    @Override
+                    public void onCancel(DialogInterface dialog) {
+                        m_finish_app = true;
+                        finish();
+                    }
+                });
+        AlertDialog dialog = builder.create();
+        dialog.show();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
```

Comment:
It's strange to do like that, but it works. I think it's maybe cleared to explicitly start the next Activity when appropriate rather than doing like that. `m_finish_app` is not necessary then